### PR TITLE
clean tests folder with make target

### PR DESCRIPTION
**Pull request by Sholde**
Hello,

This PR is related to issues #193.
It provides a **clean.sh** for all current tests.
If a test doesn't have a **clean.sh** script, then this message appear:
```
./test_name have not clean.sh script
```
Instead of:
```
./test_name OK
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,6 +122,14 @@ Then you can run the test suite with,
 If you disable flang support during configure, Fortran tests will be
 disabled and considered as passing the test.
 
+### Cleanning tests
```
Comment (pablooliveira) [inline README.md:125]: Cleanning -> Cleaning.

File: tests/clean.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+# Print number of tests detected
+echo "1..$count tests"
+
+for t in $tests ; do
+    if [ -e $t/clean.sh ] ; then
+        echo $t OK
+        bash -c -l "cd $t && ./clean.sh"
+    else
+        echo $t have not clean.sh script
```
Comment (pablooliveira) [inline tests/clean.sh:15]: "have no clean.sh" -> "does not have a clean.sh"

**Review by pablooliveira — CHANGES_REQUESTED** [review]
Looks good. Thanks for the contribution. Could you please fix the few typos ?

File: README.md (hunk 1 above)
Comment (pablooliveira) [reply README.md:125]: Actually. Can you please remove this from the documentation; I think there is no need to document explicitly this feature.

File: README.md [hunk 3]
```diff
@@ -122,6 +122,14 @@ Then you can run the test suite with,
...
+
+You can clean the test directory with,
+
+```bash
+   $ make cleantests
```
Comment (pablooliveira) [inline README.md:130]: Can we have just `make clean` do the cleaning instead of `make cleantests` ?

File: README.md (hunk 1 above)
Comment (Sholde) [reply README.md]: Yes, of course.

File: tests/clean.sh (hunk 2 above)
Comment (Sholde) [reply tests/clean.sh]: Thanks.

File: README.md (hunk 3 above)
Comment (Sholde) [reply README.md]: Yes it will be more efficient. I will try to add my cleantests target to clean target.

**Comment by pablooliveira** [comment]
Congrats. This looks good @Sholde. Will merge if tests are green.

**Review by pablooliveira — APPROVED** [review]

**Pull request merged**